### DISCUSSION TITLE
Sort shopping list by food categories

### DISCUSSION THREAD

**Discussion by Selduro** [mealie-recipes/mealie#3591] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

Having the shopping items already sorted already in the same order as the supermarket hosts them, shopping gets way more efficient because you don't have to scroll through the shopping list after every item to see, what's coming up next in the store.

### Please provide a concise description of the feature that would resolve your issue.

I think, 2 things are needed:

- Categories for foods. These could include:
  - Fruits and vegetables
  - Meat
  - Baking stuff
  - coffee and tea
  - Pasta and rice
  - dairy products
  - spices
  - beverages
  - frozen food
- Autosorting the shopping list via a customizable order of categories, so it suits your main supermarket

It's basically what this app does when you input foods [https://play.google.com/store/apps/details?id=de.markusliebschner.thatlist](https://play.google.com/store/apps/details?id=de.markusliebschner.thatlist)

### Please consider and list out some caveats or tradeoffs made in your design decision

First setup might be annoying, having to link a lot of foods to their corresponding categories.

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [X] I'm willing to sponsor/pay a developer to do this work

**Comment by boc-the-git** [comment]
Well, you'll be happy to know this already exists! :)

What you've requested sounds like an exact match for what we already have :D

Toggle to enable, then reorder to put them in the order you want. You might need to modify your data for food to set their labels.
If you need help, you'll get plenty on Discord.

![image](https://github.com/mealie-recipes/mealie/assets/3479092/f3de2b94-fb0f-4e01-b452-35a0712622c7)
